elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/client/impl/BackendRequestProcessor.java

```diff
@@ -224,6 +239,11 @@ public void shutdown() {
 			scheduler.shutdown();
 			try {
 				scheduler.awaitTermination( Long.MAX_VALUE, TimeUnit.SECONDS );
+				if ( lastAsyncWorkLatch != null ) {
+					//It's possible that a task was successfully scheduled but had no chance to run,
```

Comment:
I think you're correct, the inference of my 3rd point above is not.

What I was having in mind is that "if anything goes wrong" with that Runnable not being actually run, other threads are stuck forever. In practice I think this should only happen during an interruption so I'll move that code to handle the case.